crypto_hasher_provider.go

```diff
@@ -15,18 +15,19 @@
 
 package ionhash
 
-type cryptoHasherProvider struct {
+// CryptoHasherProvider struct for crypto hasher provider
+type CryptoHasherProvider struct {
 	IonHasherProvider
 
 	algorithm algorithm
 }
 
-// NewCryptoHasherProvider returns a new cryptoHasherProvider.
-func NewCryptoHasherProvider(algorithm algorithm) *cryptoHasherProvider {
-	return &cryptoHasherProvider{algorithm: algorithm}
+// NewCryptoHasherProvider returns a new CryptoHasherProvider.
+func NewCryptoHasherProvider(algorithm algorithm) *CryptoHasherProvider {
```

Comment:
I am currently reading up on Go practices for APIs, should we define an interface for CryptoHasher and have functions receive and return the interface type. The implementation can use the pointer (as we have now) 

What I want to try (and if I can get to it today I should be able to run a litmus test) is to see if the interfaces get us out of the cyclic dependencies and we can move the implementations in the internal folder. 


Non-blocking for this PR.